languages/thingtalk/dialogue_acts/action-confirm.js

```diff
@@ -68,7 +72,37 @@ function actionConfirmAcceptPhrase(ctx) {
     return clone.state;
 }
 
+function actionConfirmRejectPhrase(ctx) {
+    return new Ast.DialogueState(null, POLICY_NAME, 'cancel', null, []);
+}
+
+function actionConfirmChangeParam(ctx, answer) {
+    const questions = ctx.dialogueActParam || [];
+    if (answer instanceof Ast.Value) {
+        if (questions.length !== 1)
+            return null;
+        answer = new Ast.InputParam(null, questions[0], answer);
+    }
+
+    const action = C.getInvocation(ctx.state.history[0]);
+    if (!action) return null;
+
+    // don't continue with queries
+    if (action.schema._functionType === "query") return null;
```

Comment:
I'm not sure why this check? We want to say no to confirmable queries too.
Also, `_` indicates a private property. Don't use those outside of the specific class.

Reply:
Good to know about `_`. Removing the check doesn't seem to cause any issue with generation. Though, it seems a misnomer to assign `sys_confirm_action` to a confirmable query.

Reply:
It does, but having sys_confirm_action and sys_confirm_query would be worse, and also confirmable queries are only a theoretical possibility for now, because queries don't have side effects so they're always safe to run.

Reply:
Ah, makes sense!